docs/en/interfaces/cli.md

```diff
@@ -158,6 +158,116 @@ $ clickhouse-client --param_tuple_in_tuple="(10, ('dt', 10))" -q "SELECT * FROM
+clickhouse://host1:9000,host2:5000/some_database
+```
+
+The host component can either be an IP address or a host name. Put an IPv6 address in square brackets to specify it:
+
+```text
+clickhouse://[2001:db8::1234]
+```
+
+If user or/and password are not specified, default values will be used.
+If host is not specified, the default host will be used (localhost).
+If port is not specified, the default port will be used (9000).
+If database is not specified, the default database will be used.
+
+User, password, and database can be specified in the connection string either in `--user`, `--password`, `--database` command line options.
+
+The connection string must be specified in the first argument of clickhouse-client. The connection string can be combined with other [command-line-options](#command-line-options) except `--host(h)` and `--port`.
+
+### Multiple hosts {#connection_string_multiple_hosts}
+
+URI allows multiple hosts to be connected to, and the client will try to connect to those hosts using the order from URI and command line options. The hosts and ports in the URI accept comma-separated lists of values.
+
+If more than one host is supplied, or if a single host name is translated to more than one address, each host and address will be attempted one at a time until one is successful. The remaining hosts after successful connection in the list are not tried.
+
+### Percent encoding {#connection_string_uri_percent_encoding}
+
+Hosts, user name, password, database, and query parameters should be [Percent-Encoded](https://en.wikipedia.org/wiki/URL_encoding) if values contain invalid URI characters.
+
+### Examples {#connection_string_examples}
+
+Connect to localhost using port 9000 and execute the query "SELECT 1".
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --query "SELECT 1"
+```
+
+Connect to localhost using port 9000 in interactive, multiline mode.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" -m
+```
+
+Connect to localhost using port 9000 in interactive mode with the user specified in `--user` option.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --user default
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the command line option.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000" --database my_database
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the connection string.
+
+``` bash
+clickhouse-client "clickhouse://localhost:9000/my_database"
+```
+
+Connect to localhost using port 9000 in interactive mode to `my_database` database specified in the connection string and a secure connection using shorthanded 's' URI parameter.
+
+```bash
+clickhouse-client "clickhouse://localhost/my_database?s"
+```
+
+Connect to default host using default port, default user, and default database.
+
+``` bash
+clickhouse-client "clickhouse:"
```

Comment:
Interesting. Should that not be `clickhouse://` according to l. 166?

Reply:
Line 166 is wrong a bit.

'//' is acutally part of 'authority' in URI specification.

> The authority component is preceded by a double slash ("//") and is
>    terminated by the next slash ("/"), question mark ("?"), or number
>    sign ("#") character, or by the end of the URI.

"//" must be added if URI contains user_info or any host.

clickhouse:///, clickhouse:///?, clickhouse://?, clickhouse:/? , clickhouse:? are also valid URIs with empty data

Reply:
Changed documentation